src/System.Net.Http/src/System/Net/Http/Unix/CurlCallbacks.cs

```diff
@@ -414,5 +428,76 @@ private static void CheckForCompletedTransfers(SafeCurlMultiHandle multiHandle,
                 }
             }
         }
+
+        private async static void PollFunction(SafeCurlMultiHandle multiHandle)
+        {
+            while (true)
+            {
+                if (multiHandle.CancellationToken.IsCancellationRequested)
+                {
+                    return;
+                }
+
+                var count = multiHandle.FdList.Count;
+                if (count == 0)
+                {
+                    await Task.Delay(100).ConfigureAwait(false);
+                    continue;
+                }
+
+                int result = -1;
+                unsafe
+                {
+                    Interop.libc.pollfd* fds = stackalloc Interop.libc.pollfd[count];
```

Comment:
A few hundred is fine.  If it could get up into the thousands, then yeah, having a cut-off and switching to a heap allocation (then using fixed to get the ptr) would be good.